proposals/0007-live-edge.md

```diff
@@ -0,0 +1,196 @@
+- Feature Name: Live Edge
+- Start Date: 2023-02-08
+- PR: [video-dev/media-ui-extensions#7](https://github.com/video-dev/media-ui-extensions/pull/7)
+
+# Summary
+
+This is a proposal to provide well defined behaviors and an API for the "live edge" of live streaming media, including a "window" or "offset" from the live edge to account for the discrete and segmented nature of live edge updating in modern HTTP Adaptive Streaming (HAS) standards.
+
+# Motivation
+
+For live/"DVR" media, client media players will frequently have user interfaces that are specifically related to the "live edge". These include, for example:
+
+- A "live indicator", which indicates to a viewer whether or not they are currently watching at "the live edge" of available media
+- A "seek to live" button or similar, that allows a user to easily seek to "the live edge".
+
+## The problem
+
+Unfortunately, due to nature of HTTP Adaptive Streaming (HAS), the live edge cannot be represented as a simple point/moment in the media's timeline. This is for a few reasons:
+
+1. At the manifest/playlist level, the available live segments are typically known by the client by periodically re-fetching these using in-specification polling rules. These files may also be updated by the server in a discontiguous manner as segments are ready for streaming. Together, since the client does not know when segments will be added by the server, the known advertised "true live edge" will "jump" discontiguously through this process, which needs to be accounted for as a plausible "range" or "window" for what counts as the live edge.
+2. HAS provides segmented media via a client-side pull based model (most typically, e.g., a GET request), where each segment has a duration. This means that a client must first "see" the segment (via the process described above), then GET and buffer the segment, and then (eventually) play the segment, starting at its start time. Here again, this entails a discontiguous, per-segment update of the timeline, which again needs to be accounted for via a "range" or "window", rather than a discrete point.
+3. In order to avoid live edge stalls, both MPEG-DASH and HLS have a concept of a "holdback" or "delay," which inform client players that they should not attempt to fetch/play some set of segments from the end of a playlist/manifest. Luckily, this can be treated as an independent offset calculation applied to e.g. the seekable.end(0) of a media element, which can then be used as a reference for any other live edge window computation.
```

Comment:
```suggestion
3. In order to avoid live edge stalls, both MPEG-DASH and HLS have a concept of a "holdback" or "delay", which inform client players that they should not attempt to fetch/play some set of segments from the end of a playlist/manifest. Luckily, this can be treated as an independent offset calculation applied to e.g. the `seekable.end(0)` of a media element, which can then be used as a reference for any other live edge window computation.
```